src/parsley/payloads.py

```
@@ -0,0 +1,357 @@
+}
+
+
+def get_payload_type(msg_type: str) -> type[Payload] | None:
+    """Return the payload class for a message type, or None for empty payloads.
+
+    Raises ``ValueError`` for unknown message types.
+    """
+    try:
+        return _PAYLOAD_MAP[msg_type]
+    except KeyError:
+        raise ValueError(f"Unknown message type: {msg_type}")
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)